src/hotspot/share/services/diagnosticCommand.cpp

```diff
@@ -554,13 +554,25 @@ int HeapDumpDCmd::num_arguments() {
 ClassHistogramDCmd::ClassHistogramDCmd(outputStream* output, bool heap) :
                                        DCmdWithParser(output, heap),
   _all("-all", "Inspect all objects, including unreachable objects",
-       "BOOLEAN", false, "false") {
+       "BOOLEAN", false, "false"),
+  _parallel_thread_num("-parallel",
+       "parallel threads number for heap iteration. "
+       "0 use system determined number of threads, "
+       "1 use one thread, i.e., disable parallelism, "
+       "n use n threads, n must be positive.",
```

Comment:
"Number of parallel threads for heap iteration. "
       "0 means let the VM determined the number of threads. "
       "1 means use one thread, i.e. disable parallelism. "
       "n means use n threads. n must be positive.",

Reply:
Thanks for detailed review, Chris, just modified as you suggested.

Reply:
I like this approach - good idea.